docs/REST API/rest-api.md

```diff
@@ -2406,11 +2406,13 @@ the `software` table.
 | bootstrap_package       | string | query | _Available in Fleet Premium_. Filters the hosts by the status of the MDM bootstrap package on the host. Valid options are 'installed', 'pending', or 'failed'. |
 | os_settings          | string  | query | Filters the hosts by the status of the operating system settings applied to the hosts. Valid options are 'verified', 'verifying', 'pending', or 'failed'. **Note: If this filter is used in Fleet Premium without a team ID filter, the results include only hosts that are not assigned to any team.** |
 | os_settings_disk_encryption | string | query | Filters the hosts by the status of the disk encryption setting applied to the hosts. Valid options are 'verified', 'verifying', 'action_required', 'enforcing', 'failed', or 'removing_enforcement'.  **Note: If this filter is used in Fleet Premium without a team ID filter, the results include only hosts that are not assigned to any team.** |
-| populate_software     | boolean | query | If `true`, the response will include a list of installed software for each host, including vulnerability data. (Note that software lists can be large, so this may cause significant CPU and RAM usage depending on page size and request concurrency.) |
+| populate_software     | boolean/string | query | If `without_vulenrability_details`, the response will include a list of installed software for each host, including which CVEs apply to the installed software versions. `true` adds vulnerability description, CVSS score, and other details when using Fleet Premium. See notes below on performance. |
```

Comment:
@iansltx When this was just a boolean, would it be possible for users to pass in a truthy value (as opposed to explicitly `true`) to populate software? If so, I think we'll need to take that into account for backwards compatibility. Maybe if the string is "without_vulnerability_details", do the simplified version, and if it's any other string (except maybe "false") we include the full version.

If that's the case though, my gut tells me we should consider a separate parameter to go alongside the boolean, instead of changing this to a string. (Maybe `populated_software_format` or something like that?)

Reply:
Truthy values are already handled; here's the mapping:

```go
func ParseBool(str string) (bool, error) {
	switch str {
	case "1", "t", "T", "true", "TRUE", "True":
		return true, nil
	case "0", "f", "F", "false", "FALSE", "False":
		return false, nil
	}
	return false, syntaxError("ParseBool", str)
}
```

Since we error on unsupported values (if the values are provided), the string option here is backward-compatible, and I've implemented this as an exact match special case, falling back to the previous logic to make absolutely sure this is BC.

Stacking this into the same parameter was intended to avoid the potential of invalid configuration combinations, e.g. `populate_software=false&populated_software_format=full` or `populate_software=true&populated_software_format=full` on Fleet Free.